### ISSUE TITLE
Chance of collisions?

### ISSUE THREAD

**Issue by chrisjhoughton** [bibig/node-shorthash#2]
Hi there - great module, thanks for creating! 

Just a general question - I don't suppose you've done any research into the chance of collisions occurring? Would definitely prefer to use the hash generator without a database persistence layer to ensure no collisions. 

I'd love to hear your thoughts.

**Comment by matteocontrini** [comment]
An answer to this? :)

**Comment by timhuff** [comment]
So assuming the number of possible hashes is ∑(26*2+10)^n (for n=1-7), that gives 3579345993194 possible hashes. Let's call this N. This number of possibilities is expressible with log(N)/log(2) = 42 bits of information. You can use [this calculator](http://everydayinternetstuff.com/2015/04/hash-collision-probability-calculator/) to determine the probability of collision. It appears that, for random inputs, the chance of collision exceeds 50% after about 2.5 million elements. However, it appears that this library will produce sequential hashes for sequential inputs, so if that's your use case, you should be fine.

**TL;DR**: P(collision) > 50% with 2.5 million elements.

**Comment by timhuff** [comment]
Figured I might as well put this here as well. For my porpoises, 2.5 million random elements wasn't enough (even if 50% was an acceptable risk). If it's not enough for you, either, you might want to consider [xxhashjs](https://github.com/pierrec/js-xxhash). Once installed, you can run `XXH.h64(stringToHash, 0).toString(36)` and get a 12/13-character long string with 64 bits of entropy. This will result in P(collision) < 0.03% for 100 million elements and 12/13 characters is still kinda "short": `"374uz2vg91isb"`

**Comment by matteocontrini** [comment]
@timhuff thank you very much for all the details!

**Comment by timhuff** [comment]
@matteocontrini No problem. I feel this information should be front and center on any hash library's README. Hash collisions is a pretty nasty bug and, given the popularity of this library, I'd be surprised if someone hasn't been bit by this.
